imblearn/over_sampling/_smote.py

```diff
@@ -731,13 +731,12 @@ def _fit_resample(self, X, y):
             X_resampled.append(X_new)
             y_resampled.append(y_new)
 
-        if sparse.issparse(X_new):
+        if sparse.issparse(X):
```

Comment:
changed in case of some edge case where the number of samples produced is 0 and thus X_new would cause an UnboundLocalError